module/zfs/arc.c

```diff
@@ -8192,12 +8192,13 @@ l2arc_sublist_lock(int list_num)
  * when writing buffers.
  */
 static inline uint64_t
-l2arc_log_blk_overhead(uint64_t write_sz)
+l2arc_log_blk_overhead(uint64_t write_sz, l2arc_dev_t *dev)
 {
 	uint64_t blocks = write_sz >> SPA_MINBLOCKSHIFT;
 
 	return ((blocks * sizeof (l2arc_log_ent_phys_t)) +
-	    (blocks * L2ARC_LOG_BLK_HEADER_LEN / L2ARC_LOG_BLK_ENTRIES));
+	    (blocks * L2ARC_LOG_BLK_HEADER_LEN /
+	    dev->l2ad_dev_hdr->dh_log_blk_ent));
```

Comment:
We should protect against a potential divide by zero condition here.

Reply:
The device size for ZFS is 64MB. So technically this will never happen. I will add an assert and remove the `log_entries = 0`.

Reply:
Since the minimum device size is 64MB, the minimum log entries will be >3. I will add a CTASSERT and remove log_entries comparisons to 0.

Reply:
This shouldn't be an issue in the final merged version, but I did happen to hit this divide by zero because I was testing with an existing pool with an earlier version which didn't store the `dh_log_blk_ent` on disk (so it was initialized to zero in my test).

Reply:
I see, will do!